Optional plugin fails when more than one dependency has the same artifactid

Example:

```
compile 'io.reactivex:rxjava:1.2.7', optional
compile 'io.reactivex.rxjava2:rxjava:2.0.7', optional
```

Command:
`gradle generatePomFileForMavenJavaPublication --stacktrace`

Error:

```
Caused by: java.lang.IllegalArgumentException: argument type mismatch
        at nebula.plugin.extraconfigurations.OptionalBasePlugin$_configureMavenPublishPlugin_closure2$_closure5$_closure6$_closure7$_closure8$_closure9.doCall(OptionalBasePlugin.groovy:71)
        at nebula.plugin.extraconfigurations.OptionalBasePlugin$_configureMavenPublishPlugin_closure2$_closure5$_closure6$_closure7$_closure8.doCall(OptionalBasePlugin.groovy:64)
        at org.gradle.api.internal.ClosureBackedAction.execute(ClosureBackedAction.java:70)
        at org.gradle.util.ConfigureUtil.configureTarget(ConfigureUtil.java:160)
        at org.gradle.util.ConfigureUtil.configure(ConfigureUtil.java:106)
        at org.gradle.util.ConfigureUtil$1.execute(ConfigureUtil.java:123)
        at org.gradle.api.internal.UserCodeAction.execute(UserCodeAction.java:38)
        ... 91 more

```

--- a/src/main/groovy/nebula/plugin/extraconfigurations/OptionalBasePlugin.groovy
+++ b/src/main/groovy/nebula/plugin/extraconfigurations/OptionalBasePlugin.groovy
@@ -63,7 +63,7 @@ class OptionalBasePlugin implements Plugin<Project> {
                         pub.pom.withXml {
                             project.ext.optionalDeps.each { dep ->
                                 def foundDep = asNode().dependencies.dependency.find {
-                                    it.artifactId.text() == dep.name
+                                    it.groupId.text() == dep.group && it.artifactId.text() == dep.name
                                 }
 
                                 if (foundDep) {

